core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -188,39 +189,33 @@ private[spark] class CoarseMesosSchedulerBackend(
           taskIdToSlaveId(taskId) = slaveId
           slaveIdsWithExecutors += slaveId
           coresByTaskId(taskId) = cpusToUse
-          val task = MesosTaskInfo.newBuilder()
+          // Gather cpu resources from the available resources and use them in the task.
+          val (remainingResources, taskResources) =
+            partitionResources(offer.getResourcesList, "cpus", cpusToUse)
```

Comment:
These are really CPU resources right? Can you rewrite it for readability:
```
// First allocate CPU resources, then memory resources from whatever is remaining
val (remainingResources, cpuResourcesToUse) =
  partitionResources(offer.getResourcesList, "cpus", cpusToUse)
val (_, memResourcesToUse) =
  partitionResources(remainingResources, "mem", MemoryUtils.calculateTotalMemory(sc))
```
Then down there in L200 you can just do
```
  .addAllResources(cpuResourcesToUse)
  .addAllResources(memResourcesToUse)
```